lib/internal/fs/streams.js

```diff
@@ -286,7 +286,8 @@ Object.setPrototypeOf(WriteStream, Writable);
 WriteStream.prototype._final = function(callback) {
   if (typeof this.fd !== 'number') {
     return this.once('open', function() {
-      this._final(callback);
+      // Make sure _final is called after every 'open' listener
+      process.nextTick(() => this._final(callback));
```

Comment:
Would this maybe be a good reason to keep `_final` behind a nextTick, at least in a semver-patch PR? Or maybe, why does this need to be changed here but not in the other places where we use a similar pattern (e.g. net, http2)?

Reply:
That's one reason to keep `_final` behind a nextTick but there are other reasons where it's not a good idea. I don't think this becomes a relevant problem once https://github.com/nodejs/node/pull/29656 is merged.

Reply:
> but not in the other places where we use a similar pattern (e.g. net, http2)?

It should probably be changed in those places as well.